examples/imageviewer/mapview-desktop/src/desktopMain/kotlin/example/map/CalcTiles.kt

```diff
@@ -0,0 +1,51 @@
+package example.map
+
+import kotlin.math.ceil
+import kotlin.math.log2
+import kotlin.math.roundToInt
+
+fun example.map.InternalMapState.calcTiles(): List<example.map.DisplayTileAndTile> {
+    fun geoLengthToDisplay(geoLength: Double): Int {
+        return (height * geoLength * scale).toInt()
+    }
+
+    val zoom: Int = minOf(
+        example.map.Config.MAX_ZOOM,
+        maxOf(
+            example.map.Config.MIN_ZOOM,
+            ceil(log2(geoLengthToDisplay(1.0) / example.map.TILE_SIZE.toDouble())).roundToInt() - example.map.Config.FONT_LEVEL
+        )
+    )
+    val maxTileIndex: Int = example.map.pow2(zoom)
+    val tileSize: Int = geoLengthToDisplay(1.0) / maxTileIndex + 1
+    val minI = (topLeft.x * maxTileIndex).toInt()
+    val minJ = (topLeft.y * maxTileIndex).toInt()
```

Comment:
I'd suggest to rename it to X/Y or top/left (better)

Reply:
Done